src/haven/devices/detectors/xspress.py

```diff
@@ -0,0 +1,103 @@
+import asyncio
+from collections.abc import Sequence
+
+from ophyd_async.core import (
+    AsyncStatus,
+    DetectorController,
+    PathProvider,
+    SignalR,
+    StandardDetector,
+    StrictEnum,
+    TriggerInfo,
+)
+from ophyd_async.epics import adcore
+from ophyd_async.epics.core import epics_signal_rw, epics_signal_x
+
+from .area_detectors import HavenDetector, default_path_provider
+
+
+class XspressTriggerMode(StrictEnum):
+    SOFTWARE = "Software"
+    INTERNAL = "Internal"
+    IDC = "IDC"
+    TTL_VETO_ONLY = "TTL Veto Only"
+    TTL_BOTH = "TTL Both"
+    LVDS_VETO_ONLY = "LVDS Veto Only"
+    LVDS_BOTH = "LVDS Both"
+    SOFTWARE_INTERNAL = "Software + Internal"
+
+
+class XspressDriverIO(adcore.ADBaseIO):
+    def __init__(self, prefix, name=""):
+        self.trigger_mode = epics_signal_rw(XspressTriggerMode, f"{prefix}TriggerMode")
+        self.erase_on_start = epics_signal_rw(bool, f"{prefix}EraseOnStart")
+        self.erase = epics_signal_x(f"{prefix}ERASE")
+        super().__init__(prefix=prefix, name=name)
+
+
+class XspressController(DetectorController):
+    def __init__(self, driver: adcore.ADBaseIO) -> None:
+        self._drv = driver
+
+    def get_deadtime(self, exposure: float) -> float:
+        # Xspress deadtime handling
```

Comment:
is this deadtime 0.001 correct, or it's like DXPController above?
maybe we should also add the same comment here

`# No-op value. Fill this in when we know what to include`

Reply:
Good idea. I'll add that to the xspress PR: https://github.com/spc-group/haven/pull/316